src/dag/write.ts

```diff
@@ -3,39 +3,43 @@ import {chunkDataKey, chunkMetaKey, headKey, chunkRefCountKey} from './key';
 import {Read} from './read';
 import {assertMeta, Chunk, ChunkHasher} from './chunk';
 import {assertNumber} from '../asserts';
-import {assertNotTempHash, Hash} from '../hash';
+import type {Hash} from '../hash';
 
 type HeadChange = {
   new: Hash | undefined;
   old: Hash | undefined;
 };
 
 export class Write extends Read {
-  private readonly _kvw: kv.Write;
+  protected declare readonly _kvr: kv.Write;
+
   private readonly _newChunks = new Set<Hash>();
   private readonly _changedHeads = new Map<string, HeadChange>();
 
-  constructor(kvw: kv.Write, chunkHasher: ChunkHasher) {
-    super(kvw, chunkHasher);
-    this._kvw = kvw;
+  constructor(
+    kvw: kv.Write,
+    chunkHasher: ChunkHasher,
+    assertValidHash: (hash: Hash) => void,
+  ) {
+    super(kvw, chunkHasher, assertValidHash);
   }
 
   get kvWrite(): kv.Write {
-    return this._kvw;
+    return this._kvr as kv.Write;
```

Comment:
is this necessary with the `protected declare readonly _kvr: kv.Write;` above?

Reply:
This is used in a few places outside this file

Reply:
but do you need the `as kv.Write` since you've declared `_kvr` to be of type `kv.Write` above with the line `protected declare readonly _kvr: kv.Write;`

Reply:
I do not.

Reply:
Renamed to `_tx` and kept the `declare`